processing/src/main/java/io/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -191,11 +211,91 @@ private ColumnAnalysis analyzeStringColumn(
       }
     }
 
+    if (analyzingMinMax() && cardinality > 0) {
+      min = Strings.nullToEmpty(bitmapIndex.getValue(0));
+      max = Strings.nullToEmpty(bitmapIndex.getValue(cardinality - 1));
+    }
+
     return new ColumnAnalysis(
         capabilities.getType().name(),
         capabilities.hasMultipleValues(),
         size,
         analyzingCardinality() ? cardinality : 0,
+        min,
+        max,
+        null
+    );
+  }
+
+  private ColumnAnalysis analyzeStringColumn(
+      final ColumnCapabilities capabilities,
+      final StorageAdapter storageAdapter,
+      final String columnName
+  )
+  {
+    int cardinality = 0;
+    long size = 0;
+
+    Comparable min = null;
+    Comparable max = null;
+
+    if (analyzingCardinality()) {
+      cardinality = storageAdapter.getDimensionCardinality(columnName);
+    }
+
+    if (analyzingSize()) {
```

Comment:
@navis mostly because "size" is assumed by many committers to not be very useful, as the docs on it are vague enough that it is not really clear what it's supposed to be doing. Are you getting value from it? If so maybe we should also tighten up the docs & behavior & tests to make it a really useful thing.

or, we could also deprecate it.